### ISSUE TITLE
arangorestore fails with error "HTTP 500 failed" without any explanation

### ISSUE THREAD

**Issue by Yogu** [arangodb/arangodb#2326]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- 3.1 (3.1.11)

On this operating system:
- Windows, version: 10

The issue can be reproduced using this dataset: https://github.com/arangodb/example-datasets Graphs Airline Company

These are the steps to reproduce:
1) Create a database 'airlines'
2) Download the dataset
3) Start a mingw64 shell with arango in the PATH
4) Run `arangorestore --input-directory "/path/to/Graphs/AirlineCompany" --server.database "airlines"`

The following problem occurs:

```
Server version: 3.1.11
# Connected to ArangoDB 'http+tcp://127.0.0.1:8529'
# Re-creating document collection 'airports'...
Processed 0 collection(s), read 0 byte(s) from datafiles, sent 0 batch(es)
2017-02-24T02:20:23Z [14140] ERROR got error from server: HTTP 500 (Internal Server Error): ArangoError 1: failed
```

TCP dump:

```
GET /_db/airlines/_api/version HTTP/1.1
Host: 127.0.0.1
Connection: Keep-Alive
User-Agent: ArangoDB
Accept-Encoding: deflate
Authorization: Basic cm9vdDo=

HTTP/1.1 200 OK
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 60

{"server":"arango","version":"3.1.10","license":"community"}

GET /_db/airlines/_admin/server/role HTTP/1.1
Host: 127.0.0.1
Connection: Keep-Alive
User-Agent: ArangoDB
Accept-Encoding: deflate
Authorization: Basic cm9vdDo=

HTTP/1.1 200 OK
Content-Type: application/json; charset=utf-8
Server: ArangoDB
Connection: Keep-Alive
Content-Length: 42

{"role":"SINGLE","error":false,"code":200}

PUT /_db/airlines/_api/replication/restore-collection?overwrite=true&recycleIds=false&force=false 

HTTP/1.1
Host: 127.0.0.1
Connection: Keep-Alive
User-Agent: ArangoDB
Accept-Encoding: deflate
Authorization: Basic cm9vdDo=
Content-Length: 183

{"indexes":[],"parameters":{"cid":"13703922087","deleted":false,"doCompact":true,"isVolatile":false,"maximalSize":33554432,"name":"airports","type":2,"version":4,"waitForSync":false}}

HTTP/1.1 500 Internal Server Error
Server: ArangoDB
Connection: Keep-Alive
Content-Type: application/json; charset=utf-8
Content-Length: 62

{"error":true,"errorMessage":"failed","code":500,"errorNum":1}
```

There is nothing added to the log even in `debug` level.

Instead I would be expecting: Any indication of what went wrong

**Comment by jsteemann** [comment]
I can reproduce this. It was due to the dump containing a collection version number that 3.1 does not accept (`"version":4`). I have fixed the server-side restore process so that it will return a more meaningful error message to the client instead of just "failed". Additionally I have adjusted the restore process to not care about the collection version number at all. Effectively, the restore will create a new instance of the collection so it can safely ignore the collection's version number in the dump.
